### PR TITLE
Ml folder/update folder

### PR DESCRIPTION
**Pull request by petersg83**
To merge and rebase after ⚠️ https://github.com/strapi/strapi/pull/13181

- Can update the name and location of the folder and its nested folders and files

### CODE REVIEW

File: packages/core/upload/tests/admin/folder.test.e2e.js [hunk 1]
```diff
@@ -326,4 +326,192 @@ describe('Folder', () => {
...
     });
   });
+
+  describe('update', () => {
+    test('rename a folder', async () => {
+      const folder = await createFolder('folder-name', null);
+
+      const res = await rq({
+        method: 'PUT',
+        url: `/upload/folders/${folder.id}`,
```
Comment (gu-stav) [inline packages/core/upload/tests/admin/folder.test.e2e.js:336]: For assets we are using a GET param: `?id=${file.id}`. I'd like to keep this behaviour in sync between both content-types. What do you think - should we update files or folders?

File: packages/core/upload/tests/admin/folder.test.e2e.js [hunk 2]
```diff
@@ -326,4 +326,192 @@ describe('Folder', () => {
...
+          name: 'folder-a-00',
+          parent: folder0.id,
+        },
+      });
+
+      expect(res.status).toBe(400);
+      expect(res.body.error.message).toBe('name already taken');
+    });
+
+    test('cannot move a folder to a folder that does not exist', async () => {
```
Comment (gu-stav) [inline packages/core/upload/tests/admin/folder.test.e2e.js:368]: Should we also add a test to prove the fact, that you can not move a folder into one of its children (which should throw an error)?

File: packages/core/upload/tests/admin/folder.test.e2e.js (hunk 1 above)
Comment (petersg83) [reply packages/core/upload/tests/admin/folder.test.e2e.js:336]: Files should be updated then, would it be ok for you to do it the front?
Folders follow our new API standard

File: packages/core/upload/tests/admin/folder.test.e2e.js (hunk 2 above)
Comment (petersg83) [reply packages/core/upload/tests/admin/folder.test.e2e.js:368]: You're right, it's not implemented. I'll implement in another PR if that's good for you?

File: packages/core/upload/tests/admin/folder.test.e2e.js (hunk 1 above)
Comment (gu-stav) [reply packages/core/upload/tests/admin/folder.test.e2e.js:336]: I'm ok with that.

File: packages/core/upload/tests/admin/folder.test.e2e.js (hunk 2 above)
Comment (gu-stav) [reply packages/core/upload/tests/admin/folder.test.e2e.js:368]: Nice, thanks!

**Review by gu-stav — APPROVED** [review]
Nice work!

**Pull request merged**